cmd/controller/controller-runtime-main.go

```diff
@@ -42,10 +44,17 @@ type SchemeFunc func(*runtime.Scheme) error
 // ProvideFunc adds a controller to a Manager.
 type ProvideFunc func(manager.Manager) (controller.Controller, error)
 
+// ExperimentalControllers is a list of controllers that can be injected into
+// controller-runtime. When the controllers are no longer experimental they may
+// be added to the default providers list.
+var ExperimentalControllers = map[string]ProvideFunc{
+	"subscription.eventing.knative.dev": subscription.ProvideController,
+}
```

Comment:
Why only allow opting in to the new controllers? What about something like:

```golang
const controllerOptions := map[String]ProvideFunc{
  "Subscription": subscription.ProvideController,
  "OldFeeds": feed.ProvideController,
  "OldFlow": flow.ProvideController,
  "OldEventType": eventtype.ProvideController,
}

var controllers := flag.String("controllers", "", "If non-empty, run only the specified controllers")

....

selected = []string{}
if controllers == "" {
  for k, _ := range controllerOptions {
    selected = selected.append(selected, k)
  }
else {
  selected = strings.Split(controllers, ",")
}
for _, s := range selected {
  if p, ok := controllerOptions[s]; ok {
    _, err := p(mrg); err != nil {
      return err
    }
  } else {
    logger.Errorf("Unable to find controller for %q", s)
  }
}
```

This will allow us to test turning off the old controllers when we're ready to switch over to the new object model.